homeassistant/components/climate/demo.py

```diff
@@ -108,9 +123,13 @@ def fan_list(self):
         """List of available fan modes."""
         return self._fan_list
 
-    def set_temperature(self, temperature):
-        """Set new target temperature."""
-        self._target_temperature = temperature
+    def set_temperature(self, **data):
+        """Set new target temperatures."""
+        self._target_temperature = data[ATTR_TEMPERATURE]
+        if self._target_temperature_high:
+            self._target_temperature_high = data[ATTR_TARGET_TEMP_HIGH]
+        if self._target_temperature_low:
```

Comment:
Do a `get` on the keyword argument dict `data` instead. If `self._target_temperature_low` is set to None in init you will never be able to change it, with current code.